tools/client-credentials.sh

```diff
@@ -1,32 +1,37 @@
 #!/bin/bash
 set -e
```

Comment:
Would be great to make this more robust as:

```
set -o errexit
set -o nounset
set -o pipefail
```

Reply:
done, though `nounset` might have no effect in the current implementation since all variables are checked. 
Agree that it adds robustness for future changes however